nullaway/src/main/java/com/uber/nullaway/ErrorBuilder.java

```diff
@@ -302,12 +307,39 @@ boolean symbolHasSuppressWarningsAnnotation(Symbol symbol, String suppression) {
     return false;
   }
 
-  static String errMsgForInitializer(Set<Element> uninitFields) {
+  static int GetLineforField(Element uninitField, VisitorState state) {
+    Tree fieldTree = getTreesInstance(state).getTree(uninitField);
+    DiagnosticPosition position = (DiagnosticPosition) fieldTree;
+    TreePath path = state.getPath();
+    JCCompilationUnit compilation = (JCCompilationUnit) path.getCompilationUnit();
+    JavaFileObject file = compilation.getSourceFile();
+    DiagnosticSource source = new DiagnosticSource(file, null);
+    return source.getLineNumber(position.getStartPosition());
+  }
+
+  static String errMsgForInitializer(Set<Element> uninitFields, VisitorState state) {
     String message = "initializer method does not guarantee @NonNull ";
```

Comment:
Not sure if this matters for modern Java compilers but... given the amount of concatenations we are doing here, is there any advantage in using a `StringBuilder`?

Reply:
Yeah I think using `StringBuilder` is a bit cleaner, rather than all this string concatenation.

Reply:
Good idea, I will use the StringBuilder instead.